anvio/scgsdatabase.py

```diff
@@ -0,0 +1,791 @@
+
+        #self.cleandir(outpath)
+        #self.cleandir(path_diamondb)
+        sys.exit()
+        if True:
+            pathpickle_dictionnary__taxo = os.path.join(
+                self.outputdirectory, 'dictionnary__taxo_code_species.pickle')
+
+            if not os.path.exists(pathpickle_dictionnary__taxo):
+                dictionnary_level = self.make_dictionnary_level(self.taxofiles)
+                with open(pathpickle_dictionnary__taxo, 'wb') as handle:
+                    pickle.dump(dictionnary_level, handle, protocol=pickle.HIGHEST_PROTOCOL)
+            else:
+                with open(pathpickle_dictionnary__taxo, 'rb') as handle:
+                    dictionnary_level = pickle.load(handle)
+
+            dictionnary__low_ident = {}
+            for scgsfasta in os.listdir(outputsub):
+                pathfa = os.path.join(self.genes_files_directory, scgsfasta )
+                self.creatsubfa_ident(dictionnary_level, pathfa, self.outputdirectory, scgsfasta )
+
+            pathpickle_dictionnary__ident = os.path.join(
+                self.outputdirectory, 'dictionnary__low_ident.pickle')
+            with open(pathpickle_dictionnary__ident, 'wb') as handle:
+                pickle.dump(dictionnary__low_ident, handle, protocol=pickle.HIGHEST_PROTOCOL)
+
+
+    def is_database_exists(self):
+        if os.path.exists(os.path.join(self.SCG_data_dir, 'SCG-A.hmm.gz')):
+            raise ConfigError("It seems you already have SCG database installed in '%s', please use --reset flag if you want to re-download it." % self.SCG_data_dir)
+
+
+    def get_remote_version(self):
+        content = read_remote_file(self.database_url + '/VERSION')
+
+
+    def getfasta(self,pathfa):
+        fasta = u.ReadFasta(pathfa)
+        return(fasta)
+
+    def domatrix(self,taxofile):
+        """format tsv in dictonnary with code as key and liste for taxonomy for value"""
+        with open(taxofile,'r') as taxo:
+            linestaxo=taxo.readlines()
+            listspecies=[]
+            matrix={}
+            for line in linestaxo:
+                names=line.split("\t")
+                code=str(names[0])
+                taxos=names[1].split(";")
+                """for name in taxos[0:-2]:
+                    if not re.match('[a-z]__[A-Z]{1}[a-z]{1,}',name):"""
+
+                taxo= re.sub(r'_[A-Z]*$', '', taxos[-1]).rstrip()
+                leveltaxo3 = re.sub(r'(\\*[a-z])\_[A-Z]', r'\1', taxo)
+                last_taxo1 = re.sub(r'sp[0-9]{1,}$', '', leveltaxo3)
+                last_taxo=last_taxo1.rstrip(" ").replace(" ","_")
+                if not re.match('s__[A-Z]{1}[a-z]{1,}',last_taxo):
+                    continue
+                taxos[-1]=last_taxo
+
+                matrix[code]=taxos
+            return(matrix)
+
+
+
+    def creatsubfa(self,matrix_taxonomy,listerefence,pathfile,keylevel,dictionnary__pickel_taxo,dictionnary_level=None):
```

Comment:
function name makes zero sense. there are no spaces after commas in the header. there is no docstring to help others understand the purpose of it. I will not continue to list these problems about variable names, function and class names, missing docstrings, and coding style problems. pelase take a broad look at them and fix them all.